### ISSUE TITLE
code shows errors and after installing app video freezes  

### ISSUE THREAD

**Issue by rahulsharma8545** [red5pro/red5pro-android-app#40]
While running project it produces so many errors and app installs. 2 way video freezes after some time and video quality is also very bad.
![capture](https://cloud.githubusercontent.com/assets/22934485/19592742/bbbb5406-979a-11e6-86ca-c8bc846808b8.PNG)

**Comment by rahulsharma8545** [comment]
Ok, can you provide me the tentative date for this update? I have created an app in ios using your sdk. Now i need to implement the same in android but this is not working. So is there any tentative date when the update will be provided.

**Comment by beetlejesss** [comment]
You can create an app using https://github.com/red5pro/android-streaming-testbed as reference instead of this project. It contains the latest version of the Android SDK. If you are experiencing issues with two-way communication with Android, please submit a ticket through https://red5pro.zendesk.com/hc/en-us/requests/new with your device(s) and server details.

**Comment by rahulsharma8545** [comment]
This new sample is also not working. I am not able to broadcast or receive calls through this.Initially code shows so many errors. After that code runs but errors are still presented there. This is really very bad that your sdk is not working in so popular android platform.

**Comment by beetlejesss** [comment]
@rahulsharma8545 please submit a help desk ticket to https://red5pro.zendesk.com/hc/en-us/requests/new with the errors you are having. This is just one app that we are working on - if you are having issues with our other apps, please include those in your support ticket.

**Comment by VJ-Vicky** [comment]
@rahulsharma8545  I also wanted to use there SDK after my close friend me told about their awesome iOS SDK and how he is planning to deploying a app in App Store. So, I also thought that I will also create an application using Red5Pro and went to checkout there website. 

There website was very impressing and appealing after seeing so many companies using there SDK I tried to give it a go but when I tried to run there test APK I was disappointed none of there test app was not running even the one Play Store app. I tried many builds from GitHub as well but all of them crashed. 

When I told my friend about this (after some weeks) he was also very frustrated and disappointed from Red5Pro server but he had bought the license had and no other choice so he continued working on it. He tried to ask them several time how to merge two streams videos(Guest Functionality like in YouNow) but he couldn't got any reply. I tried to help him out by asking Red5-server moderator Paul Greogire, he tried his level best to help me out but this thing was out of his hand as he explained. They had some material for this matter but apparently it was incomplete and had next part missing.

After weeks of waiting and disappointment my friend came to me and asked for my help then we started working on this and one weekend we both worked 48 hours straight after this we finally created third party post processing logic and had to do the task on your own. 

We were able to complete my friend project in the end but he was 5 weeks late on deadline and he almost got fired. So, my suggestion is they are not serious about the service that they provide to their customer. Think when they are not able to solve your queries before purchasing there licence how they will treat you when you have bought a licence and made a contract to use Red5Pro server for next 12 months. They really need to work on their technical and support.

Nothing more to say, disappointed that such an old Streaming server has such bad SDK implementation and support. I may change my mind in future when they are more mature and stable but as of now I will stay away from them. I suggest you the same. 

Thank You

**Comment by rahulsharma8545** [comment]
![capture2](https://cloud.githubusercontent.com/assets/22934485/19728356/b21cffc0-9bb0-11e6-8885-c785f2f77fbf.PNG)
If i am not updating the gradle this error is showing.
After updating gradle the project shows following error.
![capture3](https://cloud.githubusercontent.com/assets/22934485/19728567/8bd02b70-9bb1-11e6-9ba9-5231fe17a314.PNG)
Is there any proper update of project sdk to work correctly.

**Comment by mrchrisallen** [comment]
@VJ-Vicky I don't think this is the most productive way to engage.

Paul isn't the guy handling our support. He's one of our lead developers and is usually heads down writing code to make all of this work. We never saw any of your support questions addressed to support@red5pro.com. If we don't hear from you this way, we can't track it and respond. In other words, we can't help you if you don't allow us to do so.

We are indeed "serious about the service that we provide to our customer", and I'm sorry you and your friend had trouble using our SDK for your app. It's a challenging thing to write extensible code that works for everyone's use cases. I actually think we are doing a great job at that, but of course we can always do better. We are a small company with limited resources, and based on feedback of many of our customer, we are able to provide the best solution out there. If you don't like what we've done and how we support you, then you should just use something else.

**Comment by HoldenMills** [comment]
@VJ-Vicky When using Android Studio version 2.2 you can get around the "UnsupportedMethodException" by going into Android Studio>Preferences>Build, Execution, Deployment>Instant Run and disable instant run.

![instantrun](https://cloud.githubusercontent.com/assets/13193297/19870443/f529cd46-9f86-11e6-9f17-748a5e146ec4.png)

**Comment by VJ-Vicky** [comment]
@mrchrisallen Hello Sir, Good Morning.
 Please check these support tickets from me and my friend [Support ticket number:- **538,** **570** and **601**], if you want I can show you the whole conversation as well. I know its not the most productive way to engage but I was disappointed.

All we wanted was help in merging two streams you had logic but it was incomplete, you developer had written its first part in march and was going to write second part but he didn't. If we had its second part then we could have completed it with much ease and we tried to explain the same thing to support but we couldn't got any help so we wrote our own logic. 

@HoldenMills Hello Holden, I don't want help in this matter its @rahulsharma8545 who wants help. Its better if he tries it out and inform you is it working or not.

**Comment by mrchrisallen** [comment]
@VJ-Vicky Hi, no problem. I totally get being frustrated and responding in a brash way. We have all been there at one point or another. :)

Yeah, I knew when Andy Shaules wrote that first part of the series there was a risk he would never get to the next blog post to complete it. Of course, he got pulled onto other priorities and actual development work on the core server, and he never got the time to finish the write up. We are about to release our new WebRTC support, so Andy is heads down on working out some codec transcoding issues, so it will still be some time until we can get to this.  I will look through the support ticket you listed and see if we can pull something together for you directly though. 

Also, it sounds like you guys figured out how to combine the streams on your own. Is that correct? If so, impressive work my friend!

**Comment by rahulsharma8545** [comment]
Hey, guys will I get any solution for my problem or not? I am still not able to run even just a sample app properly in android.
None of the provided solutions is helpful for me. I am not getting any proper solution.

**Comment by beetlejesss** [comment]
Did you try disabling instant run as Holden suggested?

**Comment by rahulsharma8545** [comment]
Yes, of course. But when I try that it was showing  further error. Jdk not found.
I tried to solve that but not found solution.

**Comment by beetlejesss** [comment]
What is the exact error that you get after you disable instant run?

**Comment by rahulsharma8545** [comment]
This was the exact error
![download_20161104_200819](https://cloud.githubusercontent.com/assets/22934485/20009674/63283a70-a2cb-11e6-9681-b5c2114f584b.png)

**Comment by beetlejesss** [comment]
Ok, well silly question then, but do you have Java SDK installed and did you set a JAVA_HOME variable?  What operating system are you developing on?  I can find instructions for you if you need them.

**Comment by rahulsharma8545** [comment]
It looks like a silly question but it is not. Because I have tried all of them.
I have already jdk installed.
I also set the JAVA_HOME variable. But problem did not solve.

**Comment by beetlejesss** [comment]
What operating system are you developing on?  This is not an error with our code at this point, it is an Android Studio error because it can't find your JAVA install. What version of the JDK did you install and how did you set your JAVA_HOME variable

**Comment by rahulsharma8545** [comment]
I an using windows 8.1 and jdk 1.8.
I will continue this tomorrow with further details.

**Comment by sqow** [comment]
Hey there @rahulsharma8545, I'm glad that disabling the Instant Run feature fixed your Red5 Pro related issue.

I would double check that your `JAVA_HOME` and `JDK_HOME` are both set (reference http://stackoverflow.com/questions/16574189/android-studio-installation-on-windows-7-fails-no-jdk-found#16574523 for more on that). These system/environment variables trip up many developers. You can always check them from your command prompt on Windows via:

```bat
SET JAVA_HOME
SET JDK_HOME
```

Luckily, since these system/environment variables trip up so many developers there are many answers out there on how to fix them. Here, for instance, is one which utilizes the Windows GUI: http://answers.microsoft.com/en-us/windows/forum/windows_8-hardware/javahome-variable-when-trying-to-use-android/f8c349ea-024e-49b0-b87e-6af37a74c8c5

Best of luck in nailing down your system configuration for Android development. 👍

**Comment by VJ-Vicky** [comment]
@mrchrisallen Yes, we did a work around for doing so. It creates a bit overhead on post-processing but that allowed to complete the project. 

We merged the streams in post processing, me and my friend are looking for a better solution whenever we get free time for merging logic so it creates less overhead on server.

**Comment by mrchrisallen** [comment]
@VJ-Vicky okay cool. yes using something like ffmpeg as a post process transcode can be heavy. Once we get the WebRTC release out I will see if Andy can get back to finishing that post on merging streams. There's definitely a better way to do it for you guys. Thanks for your patience. :)
